### PR TITLE
Add some info to User Guide on #/paths

### PR DESCRIPTION
**Pull request by mwichmann**
Mention in User Guide that `#foo` and `#/foo` are equivalent.
Proviade a trivial example that shows that Python doesn't evaluate top-relative paths.

This is a doc-only change.

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `CHANGES.txt` (and read the `README.rst`)
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: doc/user/separate.xml [hunk 1]
```diff
@@ -55,12 +55,12 @@
   and header files differ.  If you build "Controller A" first,
   then "Controller B", on the second build everything would
   have to be rebuilt, because &SCons; sees that the build
-  instructions differ, and thus the targets that depend on those
-  different instructions are not valid for the current build.
+  instructions are different to the ones used to build the
```
Comment (bdbaddog) [inline doc/user/separate.xml]: "different from those used in the previous build of those targets, and so the targets need to be rebuilt" ?

File: doc/user/separate.xml [hunk 2]
```diff
@@ -55,12 +55,12 @@
...
+  existing targets,  and thus the existing targets are invalid.
   Now when you go back and build for "Controller A",
   things have to be rebuilt from scratch again for the same reason.
-  However, if you can separate the places the output files
-  go, this problem can be avoided.
+  However, if you can separate the places the output files go,
```
Comment (bdbaddog) [inline doc/user/separate.xml]: ".. the locations of the output files, "?

File: doc/user/separate.xml (hunk 1 above)
Comment (mwichmann) [reply doc/user/separate.xml]: updated

File: doc/user/separate.xml (hunk 2 above)
Comment (mwichmann) [reply doc/user/separate.xml]: updated

Comment (mwichmann) [reply doc/user/separate.xml]: Hmmm, come to think of it, I didn't really even intend this change to the user guide to be part of this PR, it was part of a separate fiddle to variantdir things. Must have committed it without thinking. I can split it out if you'd prefer, or we can proceed here.

Comment (bdbaddog) [reply doc/user/separate.xml]: here should be fine.

**Pull request merged**